src/bp-blogs/bp-blogs-activity.php

```diff
@@ -240,7 +238,7 @@ function bp_blogs_format_activity_action_new_blog_post( $action, $activity ) {
 		$post = get_post( $activity->secondary_item_id );
 		restore_current_blog();
 
-		if ( ! empty( $post ) && ! is_wp_error( $post ) ) {
+		if ( ! empty( $post ) && is_a( $post, 'WP_Post' ) ) {
```

Comment:
I believe it's best to use `instanceof`.

```suggestion
		if ( ! empty( $post ) && $post instanceof WP_Post ) {
```